lib/Sema/CSApply.cpp

```diff
@@ -3802,7 +3818,13 @@ namespace {
     }
 
     Expr *visitPackExpansionExpr(PackExpansionExpr *expr) {
-      llvm_unreachable("not implemented for PackExpansionExpr");
+      for (unsigned i = 0; i < expr->getNumBindings(); ++i) {
+        auto *binding = expr->getBindings()[i];
+        expr->setBinding(i, visit(binding));
```

Comment:
Can `visit` add any implicit AST nodes for binding? I think it might be safer to add `cs.cacheTypes(<result>)` in this case.